PlexTraktSync role update

keldian opened the pull request:
- Plex envs are now fully applied with the .env file instead of Docker parameters
- More reliable way to retrieve a valid default value for PLEX_USERNAME
- Updated some vars to reflect recent changes in the PTS project
- Removed the settings.yml block since it's no longer required

Not sure about the directory tree and naming conventions for subtasks. I would appreciate some feedback there.

Also, I tried to get the setup printout to appear conditionally (first install only) but couldn't get that to work (`pause` and `when` don't play well together). If anyone has any tips about that...

maximuskowalski commented:
You would probably check for the existence of the .env file you are creating if I understand correctly. If file exists then no need for settings task.

See this sub task for an example.
https://github.com/saltyorg/Sandbox/blob/master/roles/sshwifty/tasks/subtasks/settings.yml

```yaml
- name: Settings | Check if `{{ sshwifty_paths_config_location | basename }}` exists
  stat:
    path: "{{ sshwifty_paths_config_location }}"
  register: sshwifty_config

- name: Settings | New `{{ sshwifty_paths_config_location | basename }}` tasks
  block:

    - name: Settings | Import default `{{ sshwifty_paths_config_location | basename }}`
      template:
        src: sshwifty.conf.json.j2
        dest: "{{ sshwifty_paths_config_location }}"
        owner: "{{ user.name }}"
        group: "{{ user.name }}"
        mode: 0775

  when: not sshwifty_config.stat.exists
  ```


Take note of the location set here https://github.com/saltyorg/Sandbox/blob/9a18bf8732d75a7e5777810052589d23686a8614/roles/sshwifty/defaults/main.yml#L25
```yml
sshwifty_paths_config_location: "{{ sshwifty_paths_location }}/config/sshwifty.conf.json"
```

keldian commented:
```
- name: Create default .env file
  copy:
    content: "# This is .env file for PlexTraktSync\n
              PLEX_BASEURL={{ plex_web_url }}\n
              PLEX_FALLBACKURL={{ 'http://localhost:' + plex_web_port }}\n
              PLEX_LOCALURL={{ plex_docker_url }}\n
              PLEX_TOKEN={{ plex_auth_token }}\n
              PLEX_USERNAME={{ plex_admin_username }}\n
              TRAKT_USERNAME="
    dest: "{{ plextraktsync_paths_location }}/.env"
    force: no
    group: "{{ user.name }}"
    owner: "{{ user.name }}"
    mode: 0664
```
The `force:no` parameter already says "don't create the file if it already exist". Wouldn't the check you're describing be redundant?

maximuskowalski commented:
I thought you wanted to conditionally display a message at the end on a first run.

keldian commented:
Ah, I see. I did try pretty much what you're describing, with a `loop` check for the existence of both .env and .pytrakt.json.
My trouble wasn't about forming the conditions themselves, but getting the `pause` to work with `when` without throwing an error. This issue is outlined here: https://github.com/ansible/ansible/issues/19966

I could use `msg` instead of `pause` and that would work with the aforementioned conditions, but the message won't look as nice.

No biggie. The printout will just display on every run.

Thanks.

maximuskowalski commented:
If you make the message part of a post install task, and that task only runs depending on a condition of being a first install it seems like that should be doing what you want? 
( you would need to set this boolean before the install I guess if that is when file is created. )

keldian commented:
Yes, I've tried that. It didn't matter at which section of the install the message task ran. As long as it was a pause task, it failed.

Anyway, what can I do about the CI / plextraktsync check failing? It's complaining that the plex_auth_token var is undefined, but the plex_auth_token role was imported and should give it a value. On my system when I run the role, plex_auth_token gets a value and the install completes error-free.

keldian commented:
Ok I went at it again and it works now for some reason. I had updated Saltbox right before that and that updated Ansible (stayed on the task for like 10 minutes), so maybe that has something to do with it, or maybe I had a brainfart the other day.  ¯\_(ツ)_/¯

Anyway, I reworked the whole thing following that success, and did some cleanup, Bailing from this one before it gets too messy with commits.
